container_opts.go

```diff
@@ -255,9 +223,19 @@ func WithSnapshotCleanup(ctx context.Context, client *Client, c containers.Conta
 	return nil
 }
 
+// WithNewSnapshot allocates a new snapshot to be used by the container as the
+// root filesystem in read-write mode
+func WithNewSnapshot(id string, i Image, opts ...snapshots.Opt) NewContainerOpts {
+	return withNewSnapshotBase(id, i, false, opts...)
+}
+
 // WithNewSnapshotView allocates a new snapshot to be used by the container as the
 // root filesystem in read-only mode
 func WithNewSnapshotView(id string, i Image, opts ...snapshots.Opt) NewContainerOpts {
+	return withNewSnapshotBase(id, i, true, opts...)
+}
+
+func withNewSnapshotBase(id string, i Image, readonly bool, opts ...snapshots.Opt) NewContainerOpts {
```

Comment:
A little doubt, function parameters are usually used as part of the function identity, and as a non-export function, is it better to change `withNewSnapshotBase` to `withNewSnapshot`

Reply:
updated